addons/mail/tests/discuss/test_discuss_attachment_controller.py

```diff
@@ -25,41 +24,23 @@ def test_attachment_allowed_upload_public_channel(self):
             ),
         )
 
-    def test_attachment_delete_linked_to_public_channel(self):
-        """Test access to delete an attachment associated with a public channel"""
+    def test_attachment_delete_linked_to_channel(self):
+        """Test access to delete an attachment associated with a channel
+        whether or not limited `ownership_token` is sent"""
         channel = self.env["discuss.channel"].create({"name": "public channel"})
+        # Subtest format: (user, token, result)
         self._execute_subtests_delete(
-            product(
-                (self.guest, self.user_portal, self.user_public),
-                (self.WITH_TOKEN, self.NO_TOKEN),
-            ),
-            allowed=False,
-            thread=channel,
-        )
-        self._execute_subtests_delete(
-            product(
-                (self.user_admin, self.user_employee),
-                (self.WITH_TOKEN, self.NO_TOKEN),
-            ),
-            allowed=True,
-            thread=channel,
-        )
-
-    def test_attachment_delete_linked_to_private_channel(self):
-        """Test access to delete an attachment associated with a private channel"""
-        channel = self.env["discuss.channel"].create(
-            {"name": "Private Channel", "channel_type": "group"}
-        )
-        self._execute_subtests_delete(
-            product(
-                (self.guest, self.user_employee, self.user_portal, self.user_public),
-                (self.WITH_TOKEN, self.NO_TOKEN),
+            (
+                (self.guest, True, True),
+                (self.guest, False, False),
+                (self.user_admin, True, True),
+                (self.user_admin, False, True),
+                (self.user_employee, True, True),
+                (self.user_employee, False, True),
+                (self.user_portal, True, True),
+                (self.user_portal, False, False),
+                (self.user_public, True, True),
+                (self.user_public, False, False),
```

Comment:
shouldn't the diff simply be to change the product, rather than re-writing the whole test to the old format?

token -> true
no_token not internal -> false
no token internal/admin -> true

Reply:
I'd keep the test for the private channel and apply a similar logic.